code/go/internal/validator/semantic/validate_agent_version.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package semantic
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/elastic/package-spec/v3/code/go/internal/fspath"
+	"github.com/elastic/package-spec/v3/code/go/internal/pkgpath"
+	"github.com/elastic/package-spec/v3/code/go/pkg/specerrors"
+)
+
+var (
+	errInvalidAgentVersionCondition = fmt.Errorf("invalid agent.version condition")
+	errAgentVersionIncorrectType    = fmt.Errorf("manifest agent version is not a string")
+)
+
+// ValidateMinimumAgentVersion checks that the package manifest includes the agent.version condition.
+func ValidateMinimumAgentVersion(fsys fspath.FS) specerrors.ValidationErrors {
+	manifest, err := readManifest(fsys)
+	if err != nil {
+		return specerrors.ValidationErrors{specerrors.NewStructuredError(err, specerrors.UnassignedCode)}
+	}
+
+	agentVersionCondition, err := getAgentVersionCondition(*manifest)
+	if err != nil {
+		return specerrors.ValidationErrors{specerrors.NewStructuredError(err, specerrors.UnassignedCode)}
+	}
+
+	if _, err := semver.NewConstraint(agentVersionCondition); agentVersionCondition != "" && err != nil {
```

Comment:
@jsoriano used here the validation of the format, if the value is empty it fails as it not compliant with semver contraint.